db/migrate/20170612115845_dedupe_embedded_specialist_change_history.rb

```diff
@@ -0,0 +1,102 @@
+class DedupeEmbeddedSpecialistChangeHistory < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+
+  def up
+    # Find ChangeNotes which have no edition id and
+    # have duplicate documents / notes, these need to be removed first.
+    # The only publishing apps affected are specialist publisher and service manual publisher.
+    dupes_sql = <<-SQL
+      SELECT max(id)
+      FROM change_notes
+      WHERE edition_id IS NULL
+      GROUP BY document_id, note
+      HAVING (count(*) > 1)
```

Comment:
I struggled to find a way to give me back all the change note ids where there was duplication (defined by grouping document and note) as the grouping clause won't allow the query to return all change note ids. Using max(id) gives me back one dupe only, by looping over this query we delete the dupes by reduction/repetition.
I attempted this by trying the grouping in a sub-query but the outer select could to my mind always return non-duplicates.
If there's a handy way in SQL of determining the dupes then giving back the ids for all but one in the series it would improve this massively.